### ISSUE TITLE
On the very first run: Failed to get image from provider: supplementaryCanvas

### ISSUE THREAD

**Issue by szotsaki** [bmeaut/grainautline#41]
After the very first image loading, Qt indicates the following error: 

```
[Warning] qrc:/PsdCanvas.qml:10:5: QML Image: Failed to get image from provider: image://canvas/supplementaryCanvas [PsdCanvas.qml:10].
```

This is probably because in `SupplementaryImageLayer` the `dirty` flag is false when the two, default-constructed mouse coordinates are the same.

**Comment by csorbakristof** [comment]
Seems to be solved, it did not re-appear during the last month.

**Comment by szotsaki** [comment]
Warning after ea057af8b4:

```
[Warning] Attempted rendering "supplementaryCanvas" with invalid cv::Mat cache [ImageProviderImage.cpp:55]
```

and

```
[Warning] Attempted rendering "supplementaryCanvas" with invalid Pixel Buffer Object [ImageProviderTexture.cpp:104]
```

The reason is the same as mentioned in the very first comment: for the first run it states that the canvas is not dirty, thus the given Cache object is not stored anywhere.

**Comment by szotsaki** [comment]
Fixed in fa3324038fcd and in fe7a7f90f9ccb.
